action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -95,6 +95,30 @@ def forward_request(base_url: str, payload: BodyPayload) -> requests.Response:
         raise NotImplementedError(f"Method {payload.method} not implemented")
 
 
+async def handle_ping_pong(
+    ws: websockets.WebSocketClientProtocol,
+    pong_queue: asyncio.Queue,
+    no_connection_queue: asyncio.Queue,
+):
+    while True:
+        await asyncio.sleep(2)
+        log.debug("️➡️ outgoing ping")
+        await ws.send("ping")
+
+        try:
+            await asyncio.wait_for(pong_queue.get(), timeout=2)
```

Comment:
So the wait between pings is actually 2 seconds + WAIT FOR PONG

Reply:
It can be up to 2 extra seconds, yes